### PR TITLE
Data transfer cannot be simply determined by NVMe opcode

### PR DESCRIPTION
**Pull request by liu-wenhua**
For set/get features, it requires further checking.

### CODE REVIEW

File: lib/nvmf/rdma.c [hunk 1]
```diff
@@ -826,6 +826,18 @@ spdk_nvmf_request_prep_data(struct spdk_nvmf_request *req)
 		req->xfer = spdk_nvme_opc_get_data_transfer(req->cmd->nvmf_cmd.fctype);
 	} else {
 		req->xfer = spdk_nvme_opc_get_data_transfer(cmd->opc);
+		if ((cmd->opc == SPDK_NVME_OPC_GET_FEATURES) ||
```
Comment (dverkamp-intel) [inline lib/nvmf/rdma.c]: This needs an extra check to verify that this is an admin queue; the admin command set opcodes overlap with the NVM command set opcodes.

File: lib/nvmf/rdma.c [hunk 2]
```diff
@@ -826,6 +826,18 @@ spdk_nvmf_request_prep_data(struct spdk_nvmf_request *req)
...
+		    (cmd->opc == SPDK_NVME_OPC_SET_FEATURES)) {
+			switch (cmd->cdw10 & 0xff) {
+				case SPDK_NVME_FEAT_LBA_RANGE_TYPE:
```
Comment (dverkamp-intel) [inline lib/nvmf/rdma.c]: Please align the case statements with the switch for consistency with the rest of the code.

File: lib/nvmf/rdma.c (hunk 1 above)
Comment (liu-wenhua) [inline lib/nvmf/rdma.c]: Good catch. Will add that check.

File: lib/nvmf/rdma.c (hunk 2 above)
Comment (liu-wenhua) [inline lib/nvmf/rdma.c]: Will do.

**Comment by liu-wenhua** [comment]
I experienced a failure with the following error message which is in function spdk_nvmf_request_prep_data.
Invalid NVMf I/O Command SGL:  Type 0x%x, Subtype 0x%x

**Comment by liu-wenhua** [comment]
I have updated the change per Daniel's request. What should I do to combine the update to my original change?

**Comment by dverkamp-intel** [comment]
I will squash and merge the patches into one commit - thanks for your contribution!

**Comment by liu-wenhua** [comment]
Thanks all for help. It's great to see my contribution be accepted.

**Pull request merged**